### PR TITLE
DAOS-12859 test: use pool and container labels (pass 3)

### PR DESCRIPTION
**Pull request by daltonbohning**
Test-tag: ContainerQueryAttributeTest ManagementServiceResilience test_bashcmd ParallelIo NvmePoolCapacity DestroyTests ListPoolsTest QueryAttributeTest RbldBasic RbldBasic RbldPoolDestroyWithIO RbldReadArrayTest RbldWithIO ObjectMetadata performance,-manual container,pr DaosVolBigIO DaosVol MacsioTest TestWithScrubberBasic TestWithScrubberFault TestWithScrubberTargetEviction dfuse datamover test_container_redundancy_factor_oclass_enforcement RbldContRfTest osa RbldCascadingFailures RbldDeleteObjects RbldReadArrayTest

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by daltonbohning** [comment]
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-13210/6/tests
server/metadata.py failure is https://daosio.atlassian.net/browse/DAOS-14517

**Comment by daltonbohning** [comment]
Failures in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13210/6/testReport/
are known https://daosio.atlassian.net/browse/DAOS-14517

File: src/tests/ftest/util/verify_perms.py [hunk 1]
```diff
@@ -260,12 +260,10 @@ def _real_w(entry_type, path):
...
         try:
-            # Write a newline to reduce likelihood of corrupting an executable file
+            # Always write a line that allows the file to be executable
             with open(path, "w", encoding='utf-8') as file:
-                # O_APPEND is not supported by DFS, so "a" is not used in open. Need seek() here
-                # to move file pointer to the end of the file.
-                file.seek(0, 2)
-                return file.write('\n') == 1
+                data = '#!/usr/bin/env bash\n'
+                return file.write(data) == len(data)
```
Comment (daltonbohning) [inline src/tests/ftest/util/verify_perms.py:266]: This change was cherry-picked from https://github.com/daos-stack/daos/commit/b33a13218078f7bcdcb0032e9927b81cd55ba3be
It should merge cleanly now

File: src/tests/ftest/dfuse/mu_perms.py [hunk 2]
```diff
@@ -369,7 +369,7 @@ def _verify(use_il, expected_il_messages, expect_der_no_perm):
             self.log.info('Verify - no perms - not using IL')
             _verify(use_il=False, expected_il_messages=0, expect_der_no_perm=False)
             self.log.info('Verify - no perms - using IL')
-            _verify(use_il=True, expected_il_messages=1, expect_der_no_perm=False)
+            _verify(use_il=True, expected_il_messages=2, expect_der_no_perm=False)
```
Comment (daltonbohning) [inline src/tests/ftest/dfuse/mu_perms.py:372]: This change was cherry-picked from https://github.com/daos-stack/daos/commit/b33a13218078f7bcdcb0032e9927b81cd55ba3be
It should merge cleanly now

**Review by shimizukko — APPROVED** [review]

**Comment by daltonbohning** [comment]
Sorry for all the force pushes! I forgot this was already in review :(
Anyway, moving to draft until the CI run is clean

**Comment by daltonbohning** [comment]
Failures in 
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-13210/34/tests
- daos_vol/bigio.py - https://daosio.atlassian.net/browse/DAOS-14517
- scrubber/aggregation.py - TBD

**Comment by daltonbohning** [comment]
Failures in https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/weekly-testing/371/tests
are known
https://daosio.atlassian.net/browse/DAOS-14517

And test_list_containers is failing in weekly:
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/weekly-testing/371/tests

**Comment by daltonbohning** [comment]
Ready for review again

**Review by shimizukko — DISMISSED** [review]

**Comment by daltonbohning** [comment]
Single test failure in https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-13210/44/tests/
must be unrelated because the only changes that affect `server/metadata.py` are print statements.

**Review by shimizukko — APPROVED** [review]

**Review by phender — APPROVED** [review]

**Comment by phender** [comment]
> Single test failure in https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-13210/44/tests/ must be unrelated because the only changes that affect `server/metadata.py` are print statements.

The failure was a test timeout while destroying containers.

**Pull request merged**